src/config/configuration.rs

```diff
@@ -0,0 +1,62 @@
+use crate::config::grpc_configuration::GrpcConfiguration;
+use crate::config::transport_strategy::TransportStrategy;
+use std::time::Duration;
+
+/// Configuration for a Momento cache client.
+///
+/// Static, versioned configurations are provided for different environments:
+/// ```
+/// use momento::config::configurations;
+///
+/// /// Use laptop for local development
+/// let developer_config = configurations::laptop::LATEST;
```

Comment:
This currently uses the standard naming scheme we have, with a file called `Configuration` representing the cache client configuration and `Configurations` holding the pre-built configs. The other clients would have have prefixed config names like `TopicConfiguration`.
The full path of a pre-built in this scheme would be `momento::config::configurations::laptop::LATEST` and a hypothetical topic config would be `momento::config::topic-configurations::laptop::LATEST`. That feels pretty clunky to me. We could get combine `Configuration` and `Configurations` and rename it after the type of client it is configuring for something like this: `momento::config::cache::laptop::LATEST` and `momento::config::topic::laptop::LATEST`.
I'm open to any naming scheme.

Reply:
keep in mind that we can use the mod.rs files to `pub use` stuff and make the namespaces look however we want. I think exporting these as `momento::configurations::laptop::latest()` is probably okay but we can let kenny or other rust folks weigh in at some point. it's fungible up until we do a 1.0 so I don't think we have to bottom out on it now.

I'm a little thrown by the `LATEST` (const) without having looked at the PR yet, I would think it'd still need to be a function as opposed to a const, but maybe I'm wrong. I think in the other SDKs mostly we make it a function when we need to support passing in a logging config, and maybe that's not an issue in rust, maybe logging is configured statically like in Java and some other languages, I don't remember :)

Reply:
I updated them to functions instead of constants.

Reply:
cool. Now that i think about it, since rust doesn't support default args or function overloads, if these sigs ever need to change then we will have to create more functions anyway, so it doesn't matter that much. but at least then they'll all be functions, so i think it's good to have changed it.